### PR TITLE
Make plots transparent

### PR DESCRIPTION
**Pull request by swertz**
... for those who have non-white beamer templates ;)

Like this it's transparent by default... Should not have any negative impact right? I can make it an option if need be?

### CODE REVIEW

**Comment by OlivierBondu** [comment]
Travis failed: it seems it affects somehow the legend font size (now bigger) and one bar in the ratio plot is reappearing....
http://i.imgur.com/JavKb3T.png
http://i.imgur.com/XQ1FSEO.png

so unexpected changes, but on the good side?

**Comment by pieterdavid** [comment]
Nice idea, I didn't even know it was possible to make them transparent... the font size is due to the automatic scaling to fit within the legend box ("uncertainty" is shorter than "uncertainties", and the longest word in the legend)

**Comment by swertz** [comment]
Ha, I didn't notice the legend entry had changed... This explains that. How come??

As for the error bar appearing, that must somehow be due to the transparency?

**Comment by blinkseb** [comment]
We don't regenerate the reference image each time. The tests on the PR changing `uncertainties` to `uncertainty` were OK because the change in the resulting image was not enough to flag the two images different. It's the same for the error bar, which were added on another PR.

Here, I guess that's the transparent background makes a different when comparing the images (see difference here: http://i.imgur.com/3M219Hn.png) here, nothing too much worrying.

However, this need to be optional and off by default. I use a dark background on my slides, and the text on the plot is black -> unreadable plots!

**Comment by swertz** [comment]
PR updated. I've added the global option `transparent-background`.
I've left the legend as transparent: this shouldn't pose any problems if the pad background is white, and is probably a good thing in cases the legend comes over the plots?

**Comment by swertz** [comment]
FYI, the checks fail because the references needed to be updated from last time, it doesn't see to be due to this PR.

**Comment by blinkseb** [comment]
Can you update the reference plots?

**Comment by swertz** [comment]
Sorry, don't know how...

**Comment by blinkseb** [comment]
Simply set https://github.com/cp3-llbb/plotIt/blob/master/test/unit_tests/tests.py#L125 to `True` and execute `run_tests.sh` in the `test` folder, it should be enough to regenerate all the images.

**Comment by blinkseb** [comment]
@swertz Can you rebase and update the reference plots so I can merge this?

File: src/plotIt.cc [hunk 1]
```diff
@@ -943,7 +951,7 @@ namespace plotIt {
     // Build legend
     TLegend legend(legend_position.x1, legend_position.y1, legend_position.x2, legend_position.y2);
     legend.SetTextFont(43);
-    legend.SetFillStyle(0);
+    legend.SetFillStyle(4000);
```
Comment (blinkseb) [inline src/plotIt.cc]: This set the legend to be always transparent? What's the difference with respect to `0`?

Comment (swertz) [reply src/plotIt.cc]: It seems to work with `0`, no idea why I changed it at that time... I'll revert that.

**Comment by blinkseb** [comment]
Thanks a lot!

**Pull request merged**